tfsdk/serve_resource.go

```diff
@@ -0,0 +1,50 @@
+package tfsdk
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+func serveGetResourceType(ctx context.Context, provider Provider, typ string) (ResourceType, diag.Diagnostics) {
```

Comment:
Drive-by question: Is there a particular reason these aren't unexported methods on `Provider`? e.g.

```go
func (p Provider) getResourceType(ctx context.Context, typ string) (ResourceType, diag.Diagnostics) {/* ... */}
```